### ISSUE TITLE
Real hardware

### ISSUE THREAD

**Issue by piranna** [runtimejs/runtime#43]
According to discussion at https://github.com/runtimejs/runtime/issues/42 seems we are pretty close to be capable to run runtime.js on real hardware, being only a matter or drivers (we only support virtio). We could start supporting [the same hardware offered by QEmu for PC platform](http://wiki.qemu.org/download/qemu-doc.html#QEMU-PC-System-emulator):

- i440FX host PCI bridge and PIIX3 PCI to ISA bridge
- Cirrus CLGD 5446 PCI VGA card or dummy VGA card with Bochs VESA extensions (hardware level, including all non standard modes).
- PS/2 mouse and keyboard
- 2 PCI IDE interfaces with hard disk and CD-ROM support
- Floppy disk
- PCI and ISA network adapters
- Serial ports
- Creative SoundBlaster 16 sound card
- ENSONIQ AudioPCI ES1370 sound card
- Intel 82801AA AC97 Audio compatible sound card
- Adlib(OPL2) - Yamaha YM3812 compatible chip
- Gravis Ultrasound GF1 sound card
- CS4231A compatible sound card
- PCI UHCI USB controller and a virtual USB hub.

The network cards it support are virtio, i82551, i82557b, i82559er, ne2k_pci, ne2k_isa, pcnet, rtl8139, e1000, smc91c111, lance and mcf_fec, being the e1000 the default one for PC platform, so we should add support for it first.

After that, we can move to other well-constrained boards like Raspberry Pi. We can (and should!) left for the future some other not so basic components like advance graphic or sound cards so we can start having servers in real world.

**Comment by iefserge** [comment]
+1 e1000. It's used in Intel servers, QEMU support would greatly simplify development.
+1 to focus on network cards. Should be enough to build network servers with runtime.

However, virtio and tcp still need some more work to make them stable and feature-complete.

**Comment by piranna** [comment]
> However, virtio and tcp still need some more work to make them stable and feature-complete.

I agree on that, real hardware support don't need to be tomorrow... :-)

**Comment by iefserge** [comment]
@piranna Unless you'd like to hack on this driver :)

**Comment by piranna** [comment]
Touchée ;-) Too much busy at this moment with job, examns and my bachelor thesis (NodeOS, in fact), but thank you in the offer :-)
